### ISSUE TITLE
ConnectionError: HTTPSConnectionPool(host='ipfs.infura.io', port=5001): Max retries exceeded with url: /api/v0/cat/QmXxkBASF92QvFFwChwZAECSRsjoMstMHJ2A9bwXuchXbA (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f6791

### ISSUE THREAD

**Issue by mbeacom** [gitcoinco/web#932]
View details in Rollbar: [https://rollbar.com/gitcoin/gitcoin/items/826/](https://rollbar.com/gitcoin/gitcoin/items/826/)

```
Traceback (most recent call last):
  File "/home/ubuntu/gitcoin/coin/app/dashboard/management/commands/sync_geth.py", line 61, in handle
    bounty = get_bounty(bounty_enum, network)
  File "/home/ubuntu/gitcoin/coin/app/dashboard/utils.py", line 197, in get_bounty
    bounty_data_str = ipfs_cat(bountydata)
  File "/home/ubuntu/gitcoin/coin/app/dashboard/utils.py", line 126, in ipfs_cat
    response = ipfs_cat_requests(key)
  File "/home/ubuntu/gitcoin/coin/app/dashboard/utils.py", line 144, in ipfs_cat_requests
    response = requests.get(url)
  File "/home/ubuntu/gitcoin/gitcoin-3/lib/python3.6/site-packages/requests/api.py", line 72, in get
    return request('get', url, params=params, **kwargs)
  File "/home/ubuntu/gitcoin/gitcoin-3/lib/python3.6/site-packages/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/ubuntu/gitcoin/gitcoin-3/lib/python3.6/site-packages/requests/sessions.py", line 508, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/ubuntu/gitcoin/gitcoin-3/lib/python3.6/site-packages/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/home/ubuntu/gitcoin/gitcoin-3/lib/python3.6/site-packages/requests/adapters.py", line 508, in send
    raise ConnectionError(e, request=request)
ConnectionError: HTTPSConnectionPool(host='ipfs.infura.io', port=5001): Max retries exceeded with url: /api/v0/cat/Qmd2zZ2P6HTeYszZXQvR44haPhy4paDopPgrkJfoVPExXq (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f6790512da0>: Failed to establish a new connection: [Errno 111] Connection refused',))Traceback (most recent call last):
  File "/home/ubuntu/gitcoin/gitcoin-3/lib/python3.6/site-packages/requests/adapters.py", line 440, in send
    timeout=timeout
  File "/home/ubuntu/gitcoin/gitcoin-3/lib/python3.6/site-packages/urllib3/connectionpool.py", line 639, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/home/ubuntu/gitcoin/gitcoin-3/lib/python3.6/site-packages/urllib3/util/retry.py", line 388, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
MaxRetryError: HTTPSConnectionPool(host='ipfs.infura.io', port=5001): Max retries exceeded with url: /api/v0/cat/Qmd2zZ2P6HTeYszZXQvR44haPhy4paDopPgrkJfoVPExXq (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f6790512da0>: Failed to establish a new connection: [Errno 111] Connection refused',))Traceback (most recent call last):
  File "/home/ubuntu/gitcoin/gitcoin-3/lib/python3.6/site-packages/urllib3/connectionpool.py", line 601, in urlopen
    chunked=chunked)
  File "/home/ubuntu/gitcoin/gitcoin-3/lib/python3.6/site-packages/urllib3/connectionpool.py", line 346, in _make_request
    self._validate_conn(conn)
  File "/home/ubuntu/gitcoin/gitcoin-3/lib/python3.6/site-packages/urllib3/connectionpool.py", line 850, in _validate_conn
    conn.connect()
  File "/home/ubuntu/gitcoin/gitcoin-3/lib/python3.6/site-packages/urllib3/connection.py", line 284, in connect
    conn = self._new_conn()
  File "/home/ubuntu/gitcoin/gitcoin-3/lib/python3.6/site-packages/urllib3/connection.py", line 150, in _new_conn
    self, "Failed to establish a new connection: %s" % e)
NewConnectionError: <urllib3.connection.VerifiedHTTPSConnection object at 0x7f6790512da0>: Failed to establish a new connection: [Errno 111] Connection refusedTraceback (most recent call last):
  File "/home/ubuntu/gitcoin/gitcoin-3/lib/python3.6/site-packages/urllib3/connection.py", line 141, in _new_conn
    (self.host, self.port), self.timeout, **extra_kw)
  File "/home/ubuntu/gitcoin/gitcoin-3/lib/python3.6/site-packages/urllib3/util/connection.py", line 83, in create_connection
    raise err
  File "/home/ubuntu/gitcoin/gitcoin-3/lib/python3.6/site-packages/urllib3/util/connection.py", line 73, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused
```

**Comment by Skyge** [comment]
@mbeacom I meet a similar question now, so I come here for some tips, but I can not view that website, `Access not allowed`.

**Comment by owocki** [comment]
hey @Skyge heads up Mark left the team.  Ill reopen this for you now tho

**Comment by Skyge** [comment]
@owocki OK, thanks for your help.

**Comment by Skyge** [comment]
I am trying run commond `docker-compose exec web python3 app/manage.py sync_geth rinkeby -20 99999999999` to import data, but it throws out an error: `ipfsapi.exceptions.ConnectionError: ConnectionError: HTTPSConnectionPool(host='ipfs.infura.io', port=5001): Max retries exceeded with url: /api/v0/version?stream-channels=true (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7ff11f9b2860>: Failed to establish a new connection: [Errno 111] Connection refused'))`

**Comment by Skyge** [comment]
I use the newest version: e5525f1d316cefcd9834f7d30f72fd063b21316f(05/04)

**Comment by Skyge** [comment]
@owocki I really need to quickly solve this problem, so any idea?

**Comment by Skyge** [comment]
According to `@SaptakS` advice, there are some cases for this phenomenon:
```
this seems like a problem with the infura server being down
it's not a problem with the codebase
it's either your internet connection is weak so there is some gateway timeout 
(which can be fixed by trying it sometime later)
or ipfs server of infura was down
or some bad ssl in your terminal side
or try a different network without proxy
or  have https_proxy set in environment: 
    like export https_proxy=<proxy_address>:<proxy:port>

try rebuilding the code.
by doing `docker-compose down` and then `docker-compose up --build`
```

**Comment by Skyge** [comment]
@owocki Although I have not solved it yet, you can close this issue now, maybe this only happens on  my PC. Thanks!

**Comment by Skyge** [comment]
@owocki ,I have solved this problem, I update the newest code 2bdfa63c8c1634831d9376c1cb9dd955aab0a5ed
